paasta_tools/cli/utils.py

```diff
@@ -1083,3 +1101,31 @@ def get_paasta_oapi_api_clustername(cluster: str, is_eks: bool) -> str:
     "eks-" prefix
     """
     return f"eks-{cluster}" if is_eks else cluster
+
+
+def get_current_ecosystem() -> str:
+    """Infer ecosystem from local-run configuration"""
+    local_run_config = load_system_paasta_config().get_local_run_config()
+    ecosystem = local_run_config["default_cluster"].split("-", 1)[-1]
+    return f"{ecosystem}prod" if ecosystem == "corp" else ecosystem
+
+
+def get_service_auth_token() -> str:
+    """Uses instance profile to authenticate with Vault and generate token for service authentication"""
+    ecosystem = get_current_ecosystem()
+    vault_client = get_vault_client(get_vault_url(ecosystem), get_vault_ca(ecosystem))
+    vault_role = load_system_paasta_config().get_service_auth_vault_role()
+    metadata_provider = InstanceMetadataProvider(
+        iam_role_fetcher=InstanceMetadataFetcher(),
```

Comment:
just curious: would we need to do any sort of env var clearing for this to work if folks have any AWS_* env vars set (or if they have a default profile or anything like that) 

(i've personally run into fun stuff like that when attempting to run things that used only the instance profile)

Reply:
As far as I know, this always calls to the instance metadata endpoint, regardless of env values.

Reply:
nice, perfect!